### PR TITLE
Add environmental variables & sampling methods to standards page

### PR DESCRIPTION
**Pull request by LouisThedroux**
## Jira tickets
- N/A

## Description of changes 
- Adds method lookup values and associated qualitative/quantitative options to standards page
- Adds environmental variables and associated qualitative/quantitative options to standards page
- Updates the /standards backend

## Testing notes
- Standards page should be visible even if logged out
- Methods/environment search boxes should filter the results on the standards page

### CODE REVIEW

**Comment by LouisThedroux** [comment]
This pull request is to replace the method-standards branch and PR

File: app/src/features/standards/DoubleStandardsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                label={''}
+                handleSpecies={(value) => {
+                  if (value) {
+                    standardsDataLoader.refresh(value);
+                  }
+                }}
+              />
+            )}
+            {/* This is te bit of code that shoes the results for search bar. Is there a way to make sure this is only showing when currentTab is species? */}
+            <SpeciesStandardsResults data={standardsDataLoader.data} isLoading={standardsDataLoader.isLoading} />
```
Comment (MacDeluca) [inline app/src/features/standards/DoubleStandardsPage.tsx]: You can pull this component into the block above.
```js
            {currentTab === 'SPECIES' && (
              <>
              <SpeciesAutocompleteField
                formikFieldName="tsn"
                label={''}
                handleSpecies={(value) => {
                  if (value) {
                    standardsDataLoader.refresh(value);
                  }
                }}
              />
             <SpeciesStandardsResults data={standardsDataLoader.data} isLoading={standardsDataLoader.isLoading} />
            </>
            )}        
```

File: api/src/services/standards-service.ts [hunk 2]
```diff
@@ -59,4 +49,29 @@ export class StandardsService extends DBService {
       measurements: response[2]
     };
   }
+
+  /**
+   * Gets environment standards
+   *
+   * @param {string} keyword
```
Comment (MacDeluca) [inline api/src/services/standards-service.ts]: Can you include some additional information about what `keyword` is?

File: api/src/services/standards-service.ts [hunk 3]
```diff
@@ -59,4 +49,29 @@ export class StandardsService extends DBService {
...
+  async getEnvironmentStandards(keyword?: string): Promise<EnvironmentStandards> {
+    const response = await this.standardsRepository.getEnvironmentStandards(keyword);
+
+    return response;
+  }
+
+  /**
+   * Gets standards for method lookups
+   *
+   * @param {string} keyword
```
Comment (MacDeluca) [inline api/src/services/standards-service.ts]: Same here

File: app/src/hooks/api/useStandardsApi.ts [hunk 4]
```diff
@@ -12,16 +12,54 @@ const useStandardsApi = (axios: AxiosInstance) => {
...
+  /**
+   * Fetch method standards
+   *
+   * @param {string} keyword
+   * @return {*}  {Promise<IGetMethodsStandardsResponse>}
+   */
+  const getMethodStandards = async (keyword?: string): Promise<IMethodStandard[]> => {
+    let url = '/api/standards/methods';
+
+    if (keyword) {
```
Comment (MacDeluca) [inline app/src/hooks/api/useStandardsApi.ts]: You might be able to use `paramsSerializer`
Check useObservationApi.ts for reference.

File: app/src/hooks/api/useStandardsApi.ts [hunk 5]
```diff
@@ -12,16 +12,54 @@ const useStandardsApi = (axios: AxiosInstance) => {
...
+  /**
+   * Fetch environment standards
+   *
+   * @param {string} keyword
+   * @return {*}  {Promise<IEnvironmentStandards>}
+   */
+  const getEnvironmentStandards = async (keyword?: string): Promise<IEnvironmentStandards> => {
+    let url = '/api/standards/environment';
+
+    if (keyword) {
```
Comment (MacDeluca) [inline app/src/hooks/api/useStandardsApi.ts]: Same here

File: app/src/interfaces/useStandardsApi.interface.ts
Comment (MacDeluca) [inline app/src/interfaces/useStandardsApi.interface.ts:1]: Can we use the types from models/standard-view.ts?

File: api/src/openapi/schemas/standards.ts [hunk 6]
```diff
@@ -0,0 +1,122 @@
+import { OpenAPIV3 } from 'openapi-types';
+
+export const EnvironmentStandardsSchema: OpenAPIV3.SchemaObject = {
+  type: 'object',
+  additionalProperties: false,
+  properties: {
```
Comment (MacDeluca) [inline api/src/openapi/schemas/standards.ts:8]: We should add some descriptions in this schema, top level / properties.

File: app/src/features/standards/components/StandardsToolbar.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  const { views, currentView, setCurrentView } = props;
+
+  return (
+    <>
+      <Typography component="legend">Data types</Typography>
+      <ToggleButtonGroup
+        orientation="vertical"
+        value={currentView}
+        onChange={(_event: React.MouseEvent<HTMLElement>, view: StandardsPageView) => {
+          if (view) {
```
Comment (MacDeluca) [inline app/src/features/standards/components/StandardsToolbar.tsx:31]: Nit: Is view ever undefined?

Comment (mauberti-bc) [reply app/src/features/standards/components/StandardsToolbar.tsx:31]: Yes, if you click on an active view, view will be undefined and no tab will be selected. Wrapping with an if check makes it so there is always a view defined.

**Comment by MacQSL** [comment]
Getting an error when searching for a species in the standards page
image

File: app/src/features/standards/components/StandardsToolbar.tsx (hunk 7 above)
Comment (MacDeluca) [reply app/src/features/standards/components/StandardsToolbar.tsx:31]: Roger. We should update the typing so it is more clear. 
```js 
view: StandardsPage | null
```

**Comment by MacQSL** [comment]
I am unable to search for a taxon on the Standards page when not logged into SIMS.

If this is intended, we should not display the species selector.

**Review by MacDeluca — APPROVED** [review]
👍🏼

**Pull request merged**